test/inductor/test_torchinductor.py

```diff
@@ -5568,8 +5567,12 @@ def fn(x):
             (torch.tensor([0, 1, 3, 4, 2, 0, 0]),),
         )
 
-        with self.assertRaises(RuntimeError):
-            torch.compile(fn)(torch.tensor([0, 0, 0, 0]))
+        # non-zero's output size is data-dependent, but torch.compile implicitly assumes
+        # that it will be constant.  This causes a RuntimeError below in normal mode,
+        # but creates an out-of-bounds memory access (segfault) in cpp_wrapper mode.
+        if not config.cpp_wrapper:
```

Comment:
This is fine for now, but can we solve the root cause?

Reply:
@desertfire Thank you for poking at this; I went back and looked again, and with some assistance from @amjames we managed to root cause the actual issue (which is different from what I initially diagnosed). See #141050.